File: 1893-check-if-all-the-integers-in-a-range-are-covered/1893-check-if-all-the-integers-in-a-range-are-covered.kt
class Solution {
    fun isCovered(ranges: Array<IntArray>, left: Int, right: Int): Boolean {
        val arr = IntArray(52){0}
        for (i in 0..ranges.size-1) {
            arr[ranges[i][0]]++
            arr[ranges[i][1]+1]--
        }
        var overlap = 0
        for (i in 1..right) {
            overlap += arr[i]
            if (i >= left && overlap === 0) {
                return false
            }
        }
        return true
    }
}

File: 1893-check-if-all-the-integers-in-a-range-are-covered/1893-check-if-all-the-integers-in-a-range-are-covered.py
class Solution:
    def isCovered(self, ranges: List[List[int]], left: int, right: int) -> bool:
        arr = [0] * 52
        for lo, hi in ranges:
            arr[lo] += 1
            arr[hi+1] -= 1
        overlaps = 0
        for i in range(1, right+1):
            overlaps += arr[i]
            if i >= left and overlaps == 0:
                return False
        return True